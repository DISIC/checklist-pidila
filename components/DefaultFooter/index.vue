<template src="../../data/layout/footer.html"/>

<script>
  export default {
    name: 'DefaultFooter'
  }
</script>

<style lang="css" scoped>
  @import "../../data/layout/layout-override.css";
</style>
