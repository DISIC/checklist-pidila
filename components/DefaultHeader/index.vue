<template src="../../data/layout/header.html"/>

<script>
  export default {
    name: 'DefaultHeader'
  }
</script>

<style lang="css" scoped>
  @import "../../data/layout/layout-override.css";
</style>
