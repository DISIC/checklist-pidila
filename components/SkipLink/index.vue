<template>
  <div class="skip-link">
    <div class="container" id="top">
      <ul>
        <li><anchor-link :anchor="'main'" :message="'Aller au contenu'"/></li>
        <li><anchor-link :anchor="'filters'" :message="'Aller aux filtres'"/></li>
      </ul>
    </div>
  </div>
</template>

<script>
  import AnchorLink from '../AnchorLink/index'

  export default {
    components: {AnchorLink},
    name: 'SkipLink',
    methods: {
      scrollTo: function (anchor) {
        this.$router.scrollTo(anchor)
      }
    }
  }
</script>
