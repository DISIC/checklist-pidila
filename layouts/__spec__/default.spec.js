import {shallowMount, config} from '@vue/test-utils'

import defaultLayout from '../default.vue'
import DefaultHeader from '../../components/DefaultHeader'
import DefaultFooter from '../../components/DefaultFooter'
import SkipLink from '../../components/SkipLink'

describe('layouts | default', () => {

  it('should match snapshot', () => {
    // when
    const wrapper = shallowMount(defaultLayout, {stubs: ['nuxt']})

    // then
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should have one header component', () => {
    // when
    const wrapper = shallowMount(defaultLayout, {stubs: ['nuxt']})

    // then
    const defaultHeaderComponents = wrapper.findAll(DefaultHeader)
    expect(defaultHeaderComponents).toHaveLength(1)
  })

  it('should have one footer component', () => {
    // when
    const wrapper = shallowMount(defaultLayout, {stubs: ['nuxt']})

    // then
    const defaultFooterComponents = wrapper.findAll(DefaultFooter)
    expect(defaultFooterComponents).toHaveLength(1)
  })

  it('should have one SkipLink component', () => {
    // when
    const wrapper = shallowMount(defaultLayout, {stubs: ['nuxt']})

    // then
    const skipLinkComponents = wrapper.findAll(SkipLink)
    expect(skipLinkComponents).toHaveLength(1)
  })
})
