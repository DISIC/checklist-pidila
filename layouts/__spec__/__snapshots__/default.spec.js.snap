// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts | default should match snapshot 1`] = `
<div>
  <skip-link-stub />
   
  <default-header-stub />
   
  <nuxt-stub />
   
  <default-footer-stub />
</div>
`;
